sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample09_ObservableEventBatchLiveTests.cs

```diff
@@ -0,0 +1,236 @@
+                {
+                    var eventBody = new BinaryData($"Event #{ index }");
+                    var eventData = new EventData(eventBody);
+                    eventData.Properties.Add("ApplicationID", index);
+
+                    if (!newbatch.TryAdd(eventData))
+                    {
+                        throw new Exception($"The event at { index } could not be added.");
+                    }
+                }
+
+                //check if event 1 is in the batch
+                var contains = false;
+                foreach (var singleEvent in newbatch.Events)
+                {
+                    contains = contains || (Int32.TryParse(singleEvent.Properties["ApplicationID"].ToString(), out Int32 id) && id == 1);
+                }
+            }
+            finally
+            {
+                await producer.CloseAsync();
+            }
+            #endregion
+        }
+        /// <summary>
+        ///   Live test of the ObservableEventBatch Class. This checks that events are successfully being added
+        ///   to both the internal and external batch.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ObservableEventBatch_LiveTest()
+        {
+            await using var scope = await EventHubScope.CreateAsync(1);
+            var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+            var eventHubName = "<< NAME OF THE EVENT HUB >>";
+            /*@@*/
+            /*@@*/ connectionString = EventHubsTestEnvironment.Instance.EventHubsConnectionString;
+            /*@@*/ eventHubName = scope.EventHubName;
+            var producer = new EventHubProducerClient(connectionString, eventHubName);
+
+            try
+            {
+                using var eventBatch = await producer.CreateBatchAsync();
+                ObservableEventDataBatch newbatch = new ObservableEventDataBatch(eventBatch);
+
+                for (var index = 0; index < 5; ++index)
+                {
+                    var eventBody = new BinaryData($"Event #{ index }");
+                    var eventData = new EventData(eventBody);
+                    eventData.Properties.Add("ApplicationID", index);
+
+                    if (!newbatch.TryAdd(eventData))
+                    {
+                        throw new Exception($"The event at { index } could not be added.");
+                    }
+                }
+
+                // check if event 1 is in the batch, check that events are observable
+                var contains = false;
+                foreach (var singleEvent in newbatch.Events)
+                {
+                    contains = contains || (Int32.TryParse(singleEvent.Properties["ApplicationID"].ToString(), out Int32 id) && id == 1);
+                }
+                Assert.IsTrue(contains);
+
+                // check count is set, this means that events were successfully added to _batch
+                Assert.IsTrue(newbatch.Count > 0);
```

Comment:
```suggestion
                Assert.That(newbatch.Count, Is.GreaterThan(0), "<< Insert message here >>");
```

There may even be an `Is.GreaterThanZero()` constraint....  I don't recall.